OutOfSchool/OutOfSchool.WebApi.Tests/Services/Database/WorkshopServicesCombinerDBTests.cs

```diff
@@ -0,0 +1,147 @@
+
+    private IWorkshopServicesCombiner service;
+
+    [SetUp]
+    public void SetUp()
+    {
+        dbContextOptions = new DbContextOptionsBuilder<OutOfSchoolDbContext>()
+            .UseInMemoryDatabase(databaseName: "OutOfSchoolTestDB")
+            .UseLazyLoadingProxies()
+            .EnableSensitiveDataLogging()
+            .Options;
+
+        dbContext = new OutOfSchoolDbContext(dbContextOptions);
+
+        workshopService = new Mock<IWorkshopService>();
+        elasticsearchSynchronizationService = new Mock<IElasticsearchSynchronizationService>();
+        mapper = TestHelper.CreateMapperInstanceOfProfileTypes<CommonProfile, MappingProfile>();
+
+        var favoriteRepository = new EntityRepositorySoftDeleted<long, Favorite>(dbContext);
+        var applicationRepository = new ApplicationRepository(dbContext);
+        var workshopStrategy = new Mock<IWorkshopStrategy>();
+        var currentUserService = new Mock<ICurrentUserService>();
+        var ministryAdminService = new Mock<IMinistryAdminService>();
+        var regionAdminService = new Mock<IRegionAdminService>();
+        var codeficatorService = new Mock<ICodeficatorService>();
+        var esProvider = new Mock<IElasticsearchProvider<WorkshopES, WorkshopFilterES>>();
+
+        notificationsConfig = new Mock<IOptions<NotificationsConfig>>();
+
+        var notificationRepository = new NotificationRepository(dbContext);
+        var loggerNotificationService = new Mock<ILogger<NotificationService>>();
+        var stringLocalizer = new Mock<IStringLocalizer<SharedResource>>();
+
+        var notificationsHub = new Mock<IHubContext<NotificationHub>>();
+        var clientsMock = new Mock<IHubClients>();
+        var clientProxyMock = new Mock<IClientProxy>();
+
+        notificationsHub.Setup(wh => wh.Clients).Returns(clientsMock.Object);
+        clientsMock.Setup(client => client.Group(It.IsAny<string>())).Returns(clientProxyMock.Object);
+
+        var notificationService = new NotificationService(
+            notificationRepository,
+            loggerNotificationService.Object,
+            stringLocalizer.Object,
+            mapper,
+            notificationsHub.Object,
+            notificationsConfig.Object);
+
+        service = new WorkshopServicesCombiner(
+            workshopService.Object,
+            elasticsearchSynchronizationService.Object,
+            notificationService,
+            favoriteRepository,
+            applicationRepository,
+            workshopStrategy.Object,
+            currentUserService.Object,
+            ministryAdminService.Object,
+            regionAdminService.Object,
+            codeficatorService.Object,
+            esProvider.Object,
+            mapper);
+
+        Seed();
+    }
+
+    [TearDown]
+    public void Dispose()
+    {
+        dbContext.Dispose();
+    }
+
+    [Test]
+    public async Task Delete_WhenCalled_CreateNotificationWithTitleInAdditionalData()
```

Comment:
(Test design) I hadn't tried proposed approach in Visual Studio, so asking you to check it.

Test seems to be overcomplicated for the 1 line of code change in `WorkshopServicesCombiner`. Could you instead of creating real `notificationService` just mock it and call `.Verify(x => x.Create(...), Times.Once)` on that mock to assert that `additionalData` dictionary contains `title` data?

Thus you could probably don't use `dbContext` at all (both performance and less code, I hope). Also `SetUp()` would be simpler.

Reply:
Done

Reply:
Thank you. One last minor thing left to improve - see other comment